File: shared-environment-variables-plugin/src/main/java/dgroomes/SharedEnvVarsToolWindowFactory.java
package dgroomes;

import com.intellij.openapi.project.Project;
import com.intellij.openapi.wm.ToolWindow;
import com.intellij.openapi.wm.ToolWindowFactory;
import com.intellij.ui.content.ContentFactory;
import org.jetbrains.annotations.NotNull;

public class SharedEnvVarsToolWindowFactory implements ToolWindowFactory {

    private final ContentFactory contentFactory;

    public SharedEnvVarsToolWindowFactory() {
        contentFactory = ContentFactory.getInstance();
    }

    @Override
    public void createToolWindowContent(@NotNull Project project, @NotNull ToolWindow toolWindow) {
        var window = new SharedEnvVarsWindow();

        var content = contentFactory.createContent(window.getRootElement(), "", false);
        toolWindow.getContentManager().addContent(content);
    }
}

File: shared-environment-variables-plugin/src/dgroomes/SharedEnvVarsToolWindowFactory.kt
package dgroomes

import com.intellij.openapi.project.Project
import com.intellij.openapi.wm.ToolWindow
import com.intellij.openapi.wm.ToolWindowFactory
import com.intellij.ui.content.ContentFactory

class SharedEnvVarsToolWindowFactory : ToolWindowFactory {

    private val contentFactory: ContentFactory = ContentFactory.getInstance()

    override fun createToolWindowContent(project: Project, toolWindow: ToolWindow) {
        val window = SharedEnvVarsWindow()
        val content = contentFactory.createContent(window.rootElement, "", false)
        toolWindow.contentManager.addContent(content)
    }
}
